tools/const_finder/test/const_finder_test.dart

```diff
@@ -165,71 +197,248 @@ void _checkNonConsts() {
+      sdkRoot: sdkRoot,
+      librariesSpec: librariesSpec,
+      verify: _checkAnnotation,
+      compiler: Compiler.dart2js,
+    ),
+  ];
   try {
-    void checkProcessResult(ProcessResult result) {
-      if (result.exitCode != 0) {
-        stdout.writeln(result.stdout);
-        stderr.writeln(result.stderr);
+    stdout.writeln('Generating kernel fixtures...');
+
+    for (final _Test test in tests) {
+      test.run();
+    }
+  } finally {
+    try {
+      for (final _Test test in tests) {
+        test.dispose();
       }
-      expect(result.exitCode, 0);
+    } finally {
+      stdout.writeln('Tests ${exitCode == 0 ? 'succeeded' : 'failed'} - exit code: $exitCode');
     }
+  }
+}
 
-    stdout.writeln('Generating kernel fixtures...');
-    stdout.writeln(consts);
+enum Compiler {
+  // Uses TFA tree-shaking.
+  aot,
+  // Does not have TFA tree-shaking.
+  dart2js,
+}
 
-    checkProcessResult(Process.runSync(dart, <String>[
-      frontendServer,
-      '--sdk-root=$sdkRoot',
-      '--target=flutter',
-      '--aot',
-      '--tfa',
-      '--packages=$packageConfig',
-      '--output-dill=$boxDill',
-      box,
-    ]));
+class _Test {
+  _Test({
+    required this.name,
+    required this.dartSource,
+    required this.sdkRoot,
+    required this.verify,
+    required this.frontendServer,
+    required this.librariesSpec,
+    required this.compiler,
+  }) : dillPath = path.join(fixtures, '$name.dill');
+
+  final String name;
+  final String dartSource;
+  final String sdkRoot;
+  final String frontendServer;
+  final String librariesSpec;
+  final String dillPath;
+  void Function(String, Compiler) verify;
+  final Compiler compiler;
 
+  final List<String> resourcesToDispose = <String>[];
+
+  void run() {
+    stdout.writeln('Compiling $dartSource to $dillPath with $compiler');
+
+    if (compiler == Compiler.aot) {
+      _compileAOTDill();
+    } else {
+      _compileWebDill();
+    }
+
+    stdout.writeln('Testing $dillPath');
+
+    verify(dillPath, compiler);
+  }
+
+  void dispose() {
+    for (final String resource in resourcesToDispose) {
+      stdout.writeln('Deleting $resource');
+      File(resource).deleteSync();
+    }
+  }
+
+  void _compileAOTDill() {
     checkProcessResult(Process.runSync(dart, <String>[
       frontendServer,
       '--sdk-root=$sdkRoot',
       '--target=flutter',
       '--aot',
       '--tfa',
       '--packages=$packageConfig',
-      '--output-dill=$constsDill',
-      consts,
+      '--output-dill=$dillPath',
+      dartSource,
     ]));
 
-    checkProcessResult(Process.runSync(dart, <String>[
-      frontendServer,
-      '--sdk-root=$sdkRoot',
-      '--target=flutter',
-      '--aot',
-      '--tfa',
+    resourcesToDispose.add(dillPath);
+  }
+
+  void _compileWebDill() {
+    final ProcessResult result = Process.runSync(dart, <String>[
+      'compile',
+      'js',
+      '--libraries-spec=$librariesSpec',
+      '-Ddart.vm.product=true',
```

Comment:
@sigmundch Does `-Ddart.vm.product=true` have any impact on this `dart compile js` command? (I thought of this as a VM-specific constant)

Reply:
It does.  The Flutter Web uses the constant within the flutter framework/engine code. They use it, for example, to choose whether certain code path is executed (e.g. debug only code).